CMake: allow only build shared/static with cmake option

rkanavath opened the pull request:
the default is now build both static and shared. This is kept unchaned.
if GEOS_BUILD_SHARED or GEOS_BUILD_STATIC is OFF, then only that build is done.

GEOS_BUILD_PACKAGED option is removed and instead a check for geos_svn_revision.h is used.

rkanavath commented:
is pull requests via github are the preffered way to send patches?

strk commented:
Filing a trac ticket might help

rkanavath commented:
done. https://trac.osgeo.org/geos/ticket/788#ticket

strk commented:
There are more changes than advertised in the subject, namely the headerfile search and the macos special install section. Could you split the PRs in topical units ?

mloskot commented on CMakeLists.txt:
```diff
@@ -66,6 +66,13 @@ if(NOT MSVC)
 endif()
 
+  option(GEOS_BUILD_STATIC
+    "Set to OFF|ON (default) to build GEOS static and shared libraries" ON)
```
Should read: "to build GEOS static libraries"

mloskot commented on CMakeLists.txt:
```diff
@@ -66,6 +66,13 @@ if(NOT MSVC)
+    "Set to OFF|ON (default) to build GEOS static and shared libraries" ON)
+
+  option(GEOS_BUILD_SHARED
+    "Set to OFF|ON (default) to build GEOS static and shared libraries" ON)
```
Should read: "to build GEOS shared libraries"

rkanavath commented:
those two are two minor changes. 

osx one is a very simple one if you look because there is already a check for 
if(NOT GEOS_ENABLE_MACOSX_FRAMEWORK)

and header check is a better one other than checking with the cmake variable.

do you really want to split them in different PR? that is more work for small contribution.

rkanavath commented:
@mloskot, fixed the doc

mloskot commented on CMakeLists.txt:
```diff
@@ -254,15 +261,13 @@ MACRO (GET_SVN_REVISION)
 endif()
 
-if (NOT GEOS_BUILD_PACKAGED)
+#geos_svn_revision.h does not exists in source or binary directory, we generate it
```
Style fix - space missing: `# geos_svn_revision.h`

mloskot commented:
@rkanavath 
> do you really want to split them in different PR? that is more work for small contribution.

It does not matter. Single PR should submit single change. Bundles of unrelated changes, no matter how trivial they are, are _big no no_.

Overall, CMake changes look fine, thanks.

rkanavath commented:
fixed style

mloskot commented:
@rkanavath Please, avoid adding more commits to a pull request. Instead, update the commit (that is also why it's best if a pull request consists of single commit):

```
git commit --amend
git push -f rkanavath svn-trunk
```

strk commented:
I'll leave the decision to Matteusz, being the CMake master.
Mat, since we're at it: any way to kill appveyor ? It makes the
automatic checking of PRs useless (it's always a false positive)

rkanavath commented:
appveyor i think can be a good thing for testing. would be nice to use cmake --build . --target in appveyor configuration and let cmake handle whatever it is msbuild, nmake, ninja!

strk commented:
would be nice if it didn't fail, if it fails it defeat the purpose of
testing (it's been failing since injection:
https://lists.osgeo.org/pipermail/geos-devel/2015-October/007282.html)

On Fri, Sep 2, 2016 at 12:11 PM, Rashad Kanavath <notifications@github.com>
wrote:

> appveyor i think can be a good thing for testing. would be nice to use
> cmake --build . --target in appveyor configuration and let cmake handle
> whatever it is msbuild, nmake, ninja!
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/libgeos/libgeos/pull/66#issuecomment-244337301>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AARrjrYfVi6XYR_dSzBv09wWVrbECTk0ks5ql_ZBgaJpZM4JDL_Y>
> .
>

strk commented:
Since you're at it, include the trac ticket number in the commit log

rkanavath commented:
okay. updated commit message with trac id. removed check for header geos_revision.h 

only thing there is "extra" is the check for GEOS_ENABLE_MACOSX_FRAMEWORK. have a look at src/CMakeLists.txt and see the mutiple check there.

mloskot commented:
On 2 September 2016 at 12:07, Sandro Santilli <notifications@github.com>
wrote:

> I'll leave the decision to Matteusz, being the CMake master.
>

Please, if you have free cycles, merge it.
I won't be able to do it before next week.


> Mat, since we're at it: any way to kill appveyor ?
>

I will try to look at it again next week.

strk commented:
r4252 = 684349d20c47a4ff9e87127836ef6939f45d7f42 (refs/remotes/trunk)
